### ISSUE TITLE
Are there any plans to have a package for Golang? (I'd love to help with that too)

### ISSUE THREAD

**Issue by dhanielsales** [tursodatabase/limbo#506]
Hi there,

First, thank you for starting work on Limbo—it sounds like an exciting project!

I was wondering if there are any plans to develop a package for Golang? If such a package isn’t currently in the roadmap, I’d be thrilled to contribute to making it happen. I have experience in Golang and would be happy to collaborate. Let me know if this aligns with your goals for the project and how I could get involved.

Looking forward to your thoughts!

Cheers

**Comment by penberg** [comment]
Hey @dhanielsales, we absolutely want Limbo to be packaged for Go, but I am not aware of anyone actively working on it. 

As Limbo implements a subset of the SQLite C API, https://github.com/mattn/go-sqlite3, for example, should be straight-forward to port.

Of course, some Go folks dislike cgo so much they did:

https://github.com/ncruces/go-sqlite3

Curious to hear from you and other Gophers what's the best strategy to make this happen! And absolutely happy to take contributions here either in `bindings/go` or as an external package!

**Comment by Tejasmadhukar** [comment]
Hello there, building this project would be awesome!

Here is a benchmark for different sqlite3 drivers in go - https://github.com/cvilsmeier/go-sqlite-bench, this includes both cgo and pure go drivers along non database/sql drivers.

There is not a clear winner, this design choice/decision would be upon you.

The only thing I would like to add is the obvious fact that it would be better to have this as database/sql driver as it's defacto in go and would allow for drop in replacements in most projects.

Looking forward to hearing your thoughts and to work on this!!

**Comment by PThorpe92** [comment]
I'm on team `avoid CGO`, and I would likely contribute to this as well :+1:

**Comment by dhanielsales** [comment]
Yeah, I'm on avoid CGO team as well, so cound on me to help with it!

**Comment by nikoksr** [comment]
+1 for pure Go. Package management and distribution is much smoother without cgo dependencies in my experience. Happy to chip in on the implementation!

**Comment by petern48** [comment]
also interested in contributing!

**Comment by dvbnrg** [comment]
Count me in to contributing!

**Comment by PThorpe92** [comment]
For anyone who mentioned they were interested in contributing, I have got the Go bindings to a decent place now where contributions from others would make sense 👍 

Still lots of work to do but no longer super volatile and subject to sweeping frequent changes, so PR's are very welcome :D

**Comment by jnesss** [comment]
Progress today for Golang users!  To use the limbo adapter in your Go program, now you can:

$ git clone https://github.com/tursodatabase/limbo
$ cd limbo/bindings/go
$ ./build_lib.sh

No need to change LD_LIBRARY env variables or "go mod edit" anymore.  

Try out the latest code and leave feedback here.  I envision one day having pre-compiled libs checked in so we can just "go get..." or "go mod tidy" but until then this should make things easier.

**Comment by penberg** [comment]
There is a Go package for Limbo now. It needs more work, of course, and better packaging, but let's open specific issues for such improvements.
